package/lean/xray/Makefile

```diff
@@ -253,8 +252,8 @@ define Build/Prepare
 
 ifneq ($(CONFIG_XRAY_EXCLUDE_ASSETS),y)
 	# move file to make sure download new file every build
-	mv -f $(DL_DIR)/$(GEOIP_FILE) $(PKG_BUILD_DIR)/release/config/geoip.dat
-	mv -f $(DL_DIR)/$(GEOSITE_FILE) $(PKG_BUILD_DIR)/release/config/geosite.dat
```

Comment:
Keep them as-is to follow upstream.

Reply:
> Keep them as-is to follow upstream.

This folder does not exist in upstream

Reply:
**The source code has clearly indicated the search location,
There is no need to modify it. It has been tested and confirmed that it can be loaded normally,
Relevant Luci apps have been compatible.**

[XTLS/Xray-core](https://github.com/XTLS/Xray-core/blob/main/common/platform/others.go)
```
// GetAssetLocation search for `file` in certain locations
func GetAssetLocation(file string) string {
	const name = "xray.location.asset"
	assetPath := NewEnvFlag(name).GetValue(getExecutableDir)
	defPath := filepath.Join(assetPath, file)
	for _, p := range []string{
		defPath,
		filepath.Join("/usr/local/share/xray/", file),
		filepath.Join("/usr/share/xray/", file),
	} {
		if _, err := os.Stat(p); os.IsNotExist(err) {
			continue
		}

		// asset found
		return p
	}

	// asset not found, let the caller throw out the error
	return defPath
}
```

```
root@OpenWrt:~# ll /usr/share/xray
drwxr-xr-x    2 root     root          3488 Dec  2 18:39 ./
drwxr-xr-x    1 root     root          3488 Dec  2 18:54 ../
-rw-r--r--    1 root     root       5234261 Dec  2 14:33 geoip.dat
-rw-r--r--    1 root     root        947674 Dec  2 14:33 geosite.dat
root@OpenWrt:~# xray -config /var/v2-ssr-retcp.json
Xray 1.0.0 (OpenWrt) Lean (go1.15.5 linux/amd64)
A unified platform for anti-censorship.
2020/12/02 10:54:24 [Info] github.com/xtls/xray-core/v1/main/jsonem: Reading config: /var/v2-ssr-retcp.json
2020/12/02 10:54:24 [Warning] github.com/xtls/xray-core/v1/core: Xray 1.0.0 started
root@OpenWrt:~# ll /usr/share/xray
ls: /usr/share/xray: No such file or directory
root@OpenWrt:~# xray -config /var/v2-ssr-retcp.json
Xray 1.0.0 (OpenWrt) Lean (go1.15.5 linux/amd64)
A unified platform for anti-censorship.
2020/12/02 10:54:44 [Info] github.com/xtls/xray-core/v1/main/jsonem: Reading config: /var/v2-ssr-retcp.json
Filed to start: main: failed to read config files: [/var/v2-ssr-retcp.json] > github.com/xtls/xray-core/v1/infra/conf: invalid field rule > github.com/xtls/xray-core/v1/infra/conf: failed to parse domain rule: geosite:category-ads-all > github.com/xtls/xray-core/v1/infra/conf: failed to load geosite: CATEGORY-ADS-ALL > github.com/xtls/xray-core/v1/infra/conf: failed to open file: geosite.dat > open /usr/bin/geosite.dat: no such file or directory
```

Reply:
> failed to open file: geosite.dat > open /usr/bin/geosite.dat: no such file or directory

Keep all data files and main program in a directory, following the standard from v2fly:
All files should be stored in `/usr/bin/xray/*`

Reply:
I think you should carefully look at line 29 in the source code.  Find 3 locations, the current directory, /usr/local/share/xray/ and /usr/share/xray/

https://github.com/XTLS/Xray-core/blob/main/common/platform/others.go